spring-integration-core/src/main/java/org/springframework/integration/store/SimpleMessageStore.java

```diff
@@ -162,6 +159,7 @@ public void setCopyOnGet(boolean copyOnGet) {
 		this.copyOnGet = copyOnGet;
 	}
 
+	@Override
 	public void setLockRegistry(LockRegistry lockRegistry) {
```

Comment:
We don't need this method since it is there in the super class.

Reply:
`isUsed` is the private variable of SimpleMessageStore.

Reply:
Well, I don't think that `Assert.isTrue(!(this.isUsed), "Cannot change the lock registry after the store has been used");` really makes sense.
An unofficial rule for Spring beans is to have them setup only once during configuration phase.
So, calling setters at runtime is a bit like making a trap for yourself.
Therefore having such a big refactoring in this `MessageStore` hierarchy, I would say let's just remove this `isUsed` altogether!